From ba37bd40be4318b113f70e7161e10f72cfc914f9 Mon Sep 17 00:00:00 2001
From: Gowtham Anandha Babu <gowtham.anandha.babu@intel.com>
Date: Wed, 20 Jul 2022 14:15:39 +0530
Subject: [PATCH] Enable bluetooth profiles with product.prop

1) Set default the Bluetooth class of device

The class of device is a string with a list of uint8t values:
90,2,12

The meaning is as follows:
{Service Field, Major class, Minor class}

Service Field: 0x5A -> 90
    Bit 17: Networking
    Bit 19: Capturing
    Bit 20: Object Transfer
    Bit 22: Telephony
MAJOR_CLASS: 0x02 -> 2 (Phone)
MINOR_CLASS: 0x0C -> 12 (Smart Phone)

2) Enable bluetooth profiles with product.prop

Change-Id: Icbe1acc31522361eeaef3adebc84d23330c4e0d4
Tracked-On: OAM-102879
Signed-off-by: Gowtham Anandha Babu gowtham.anandha.babu@intel.com
---
 groups/device-specific/caas/system.prop | 32 +++++++++++++++++++++++++
 1 file changed, 32 insertions(+)

diff --git a/groups/device-specific/caas/system.prop b/groups/device-specific/caas/system.prop
index 847138a..dff1c6d 100644
--- a/groups/device-specific/caas/system.prop
+++ b/groups/device-specific/caas/system.prop
@@ -11,3 +11,35 @@ audio.safemedia.bypass=true
 debug.nn.cpuonly=0
 camera.disable_treble=false
 ro.incremental.enable=yes
+
+# Set the Bluetooth Class of Device
+# Service Field: 0x5A -> 90
+#    Bit 17: Networking
+#    Bit 19: Capturing
+#    Bit 20: Object Transfer
+#    Bit 22: Telephony
+# MAJOR_CLASS: 0x02 -> 2 (Phone)
+# MINOR_CLASS: 0x0C -> 12 (Smart Phone)
+bluetooth.device.class_of_device=90,2,12
+
+# Set supported Bluetooth profiles to enabled
+bluetooth.profile.asha.central.enabled=true
+bluetooth.profile.a2dp.source.enabled=true
+bluetooth.profile.avrcp.target.enabled=true
+bluetooth.profile.bap.broadcast.assist.enabled=true
+bluetooth.profile.bap.unicast.client.enabled=true
+bluetooth.profile.bas.client.enabled=true
+bluetooth.profile.csip.set_coordinator.enabled=true
+bluetooth.profile.gatt.enabled=true
+bluetooth.profile.hap.client.enabled=true
+bluetooth.profile.hfp.ag.enabled=true
+bluetooth.profile.hid.device.enabled=true
+bluetooth.profile.hid.host.enabled=true
+bluetooth.profile.map.server.enabled=true
+bluetooth.profile.mcp.server.enabled=true
+bluetooth.profile.opp.enabled=true
+bluetooth.profile.pan.nap.enabled=true
+bluetooth.profile.pan.panu.enabled=true
+bluetooth.profile.pbap.server.enabled=true
+bluetooth.profile.ccp.server.enabled=true
+bluetooth.profile.vcp.controller.enabled=true
-- 
2.17.1

